packages/@tinacms/cli/src/cmds/start-server/index.ts

```diff
@@ -105,11 +109,16 @@ stack: ${code.stack || "No stack was provided"}`)
   const start = async () => {
     const s = require('./server')
     state.server = await s.default(experimental)
+
     state.server.listen(port, () => {
       logger.info(`Started Filesystem GraphQL server on port: ${port}`)
       logger.info(`Visit the playground at http://localhost:${port}/altair/`)
     })
-    state.server.on('connection', socket => {
+    state.server.on('error', function () {
```

Comment:
instead of doing this for all errors can we only display the error message on a more specific code.

```ts
state.server.on('error', function (e) {
      if (e.code === 'EADDRINUSE') {
        // If port is already in use warn the user and fail
        logger.error(
          dangerText(
            `ERROR: Can not start server. Port 4001 already in use. Please kill other processes and try again.`
          )
        )
        process.exit()
      }
      throw e
    })
```

Reply:
Yes, of course. I'm sorry, I don't know how I missed that 😅